plugins/catalog-backend/src/next/DefaultCatalogProcessingEngine.ts

```diff
@@ -110,56 +110,62 @@ export class DefaultCatalogProcessingEngine implements CatalogProcessingEngine {
       return;
     }
 
-    const { id, state, unprocessedEntity } = items[0];
-
-    const result = await this.orchestrator.process({
-      entity: unprocessedEntity,
-      state,
-    });
-    for (const error of result.errors) {
-      this.logger.warn(error.message);
-    }
-    const errorsString = JSON.stringify(
-      result.errors.map(e => serializeError(e)),
-    );
-
-    // If the result was marked as not OK, it signals that some part of the
-    // processing pipeline threw an exception. This can happen both as part of
-    // non-catastrophic things such as due to validation errors, as well as if
-    // something fatal happens inside the processing for other reasons. In any
-    // case, this means we can't trust that anything in the output is okay. So
-    // just store the errors and trigger a stich so that they become visible to
-    // the outside.
-    if (!result.ok) {
-      await this.processingDatabase.transaction(async tx => {
-        await this.processingDatabase.updateProcessedEntityErrors(tx, {
-          id,
-          errors: errorsString,
+    await Promise.all(
```

Comment:
Noting that we want to replace this with something that provides high parallelization even if processing times vary a lot. Could perhaps add a todo but I thinking we'll have a look at this pretty soon so could :shipit: instead too :>

Reply:
Yeah to make it more of a "streaming" thing with a max concurrency, I think we'd want to rewrite things in the main loop pretty significantly. We could do that later